presto-password-authenticators/src/main/java/io/prestosql/plugin/password/filebasedauthenticator/User.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.password.filebasedauthenticator;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.cache.Cache;
+import io.airlift.log.Logger;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.StandardErrorCode;
+
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static io.prestosql.plugin.password.filebasedauthenticator.EncryptionHelper.checkAndGetHashing;
+import static io.prestosql.plugin.password.filebasedauthenticator.EncryptionHelper.checkBCryptPassword;
+import static io.prestosql.plugin.password.filebasedauthenticator.EncryptionHelper.checkPBKDF2Password;
+import static java.util.Objects.requireNonNull;
+
+public class User
+{
+    private final String name;
+    private final String password;
+    private static final Logger log = Logger.get(User.class);
+    private Optional<HashingAlgorithm> hashingAlgo = Optional.empty();
+    private Cache<PasswordStore, Boolean> authCacheStore;
```

Comment:
Avoid mixing pure data classes like `User` with classes that have behavior (like caching). Instead, make `User` and `PasswordStore` simple immutable classes, and create a new class `PasswordValidator` or similar to handle the logic and caching.

Reply:
Done